File: 1797-goal-parser-interpretation/1797-goal-parser-interpretation.cpp
class Solution {
public:
    string interpret(string command) {
        string parsed = "";
        for(int i=0; i < command.size(); i++ ){
            if (command[i]  == 'G'){
                parsed += 'G';
            }else if (command[i] == '('){
                if (command[i+1] == ')'){
                    parsed += 'o';
                    i++;
                }
            }else{
                parsed += "al";
                i += 2;
            }
        }
        return parsed;
    }
};

File: 1797-goal-parser-interpretation/1797-goal-parser-interpretation.py
class Solution:
    def interpret(self, command: str) -> str:
        ans = ""
        i = 0
        while i < len(command)-1:  
            if command[i] == 'G': 
                ans += 'G'
                i+=1
            else:
                if command[i+1] == ')': 
                    ans += 'o'
                    i+=2
                else:
                    ans += 'al'
                    i+= 4
        if command[len(command)-1] == "G": 
            ans += 'G'
        return ans 